gnucash/gnome/window-reconcile.c

```diff
@@ -385,14 +386,35 @@ gnc_start_recn_date_changed (GtkWidget *widget, startRecnWindowData *data)
     gnc_numeric new_balance;
     time64 new_date;
 
-    GtkWidget *future_box;
-    gboolean date_is_future;
+    GtkWidget *future_icon, *future_text;
+    gint days_after_today;
 
     new_date = gnc_date_edit_get_date_end (gde);
 
-    date_is_future = (new_date > gnc_time64_get_today_end ());
-    future_box = data->future_box;
-    gtk_widget_set_visible (future_box, date_is_future);
+    days_after_today = (gint) ((gnc_time64_get_day_end (new_date) -
+                                gnc_time64_get_today_end ()) / 86400);
+    future_icon = data->future_icon;
+    future_text = data->future_text;
+    if (days_after_today > 0)
+    {
+        gchar *str = g_strdup_printf (_("Statement Date is %u days after today."),
+                                      days_after_today);
+
+        gchar *tip = g_strdup_printf ("Note the statement date you have \
+chosen is %u days in the future. This may cause issues for future \
+reconciliation actions on this account. Please double-check this is \
+the date you intended.", days_after_today);
+
+        gtk_label_set_text (GTK_LABEL (future_text), str);
+        gtk_widget_set_tooltip_text (future_text, tip);
+        gtk_widget_show (future_text);
+        gtk_widget_show (future_icon);
+    }
+    else
+    {
+        gtk_widget_hide (future_text);
+        gtk_widget_hide (future_icon);
+    }
```

Comment:
change from 400 to this...

        /* Translators: This is a ngettext(3) message, %d is the number of days in the future */
        gchar *str = g_strdup_printf (ngettext ("Statement Date is %d day after today.",
                                                "Statement Date is %d days after today.",
                                                days_after_today),
                                                days_after_today);

         gchar *tip_end = g_strdup (_("This may cause issues for future reconciliation actions on this account. \
 Please double-check this is the date you intended."));

        /* Translators: This is a ngettext(3) message, %d is the number of days in the future */
        gchar *tip_start = g_strdup_printf (ngettext ("The statement date you have chosen is %d day in the future.",
                                                      "The statement date you have chosen is %d days in the future.",
                                                      days_after_today),
                                                      days_after_today);

        gchar *tip = g_strdup_printf ("%s\n%s", tip_start, tip_end);

        hide_warning = FALSE;

        gtk_label_set_text (GTK_LABEL(data->future_text), str);
        gtk_widget_set_tooltip_text (GTK_WIDGET(data->future_text), tip);
        g_free (str);
        g_free (tip_end);
        g_free (tip_start);
        g_free (tip); 
    }
    gtk_widget_set_visible (GTK_WIDGET(data->future_icon), !hide_warning);
    gtk_widget_set_visible (GTK_WIDGET(data->future_text), !hide_warning);